### ISSUE TITLE
When I execute my code in a specific physical location, an error occurs.

### ISSUE THREAD

**Issue by takhyun12** [spatialaudio/python-sounddevice#326]
Hi,
I am currently using SoundDevice for handle audio data.

My code works perfectly, but when I execute my code in a specific physical location an error occurs. 
Crucially, if I move elsewhere, the problem is solved.

This is my code:

```
import time
import sounddevice as sd

class Test:
    def __init__(self):
        self.block_shift = 128  # audio block shift
        self.block_len = 512  # audio block length
        self.sampling_rate = 16000  # sampling rate
        self.sound_device_latency = 'low'  # latency of callback

    def Start(self, input_device_index, output_device_index):
        with sd.Stream(device=(input_device_index, output_device_index),latency=self.sound_device_latency,
                   channels=1, callback=self.callback) as stream_object:
            while stream_object.active:
                time.sleep(0.1)

    def callback(self, indata, outdata, frames, time, status):
        if status:
            print(status)
        #print(indata)
        outdata[:] = indata

Test().Start(2, 3)
```

When I execute my code in my office, the input data (indata) passed from the microphone becomes [0.]. But there is no problem in cafes, restaurants or my house.

When I print `indata` and `status` in the `callback`, it is as follows :
```
input overflow, output underflow (status)
[0.] (indata)
[0.] (indata)
[0.] (indata)
...
output underflow (status)
[0.] (indata)
[0.] (indata)
[0.] (indata)
...
input overflow (status)
input overflow (status)
input overflow (status)
input overflow (status)
input overflow (status)
...
```

My computer's CPU and memory are fine.

Please review my code and give me some advice.
Thank you so much.

**Comment by mgeier** [comment]
See also https://stackoverflow.com/q/67190587/.

**Comment by takhyun12** [comment]
Glad to see your reply :)
I found the cause of this problem. Does `sounddevice` support anti-aliasing filter?

In my opinion, it is not a security issue, it is estimated to be an inaudible frequency problem.

**Comment by mgeier** [comment]
The Python module `sounddevice` doesn't do any sample rate conversion on its own, therefore it doesn't need an anti-aliasing filter.

The underlying PortAudio library, however, does sample rate conversion in some cases, and when it downsamples, I'm sure it uses an anti-aliasing filter.

> it is estimated to be an inaudible frequency problem

What kind of frequencies are you trying to record?

Are your `indata` values actually `0.0` or are they just very tiny numbers close to zero?

Even if your high-frequency sound is somehow filtered out, you should still expect to record some background noise.

If the recorded values are always `0.0`, this is a sign that the operating system somehow blocks the audio device and doesn't allow recording anything.

**Comment by takhyun12** [comment]
First of all, thank you so much for your help.

I used downsamples the frequencies input through the microphone to `16,000`.

If I start recording, a value of `0.0` also occurs, but not every time.

I'm in my completely quiet office, and when I check the results after recording, there's a strange sound recorded.

Also, suddenly CPU utilization increases for no reason, resulting in `input overflow`, `output underflow`. (For me, This problem is the most serious.)

I used a microphone built into my MacBook for recording.

But Surprisingly, if I take out my MacBook from my office, the problem is solved neatly.

If I 10 meter away from the office, this problem will not occurred.

I decided to call the frequency inspection company and inspect my office.

I'll let you know again when the problem is solved today.

Once again, thank you very much for your help.

**Comment by mgeier** [comment]
Any news on this?

**Comment by takhyun12** [comment]
Hi.
sorry, I'm late.

First of all, I received a report that there was a problem from the Professional frequency company I called.

This problem is that there are too many unknown frequencies in my office, and the frequency meter also worked strangely.

but we couldn't find where the frequency came from.

I figured something out, though.

The `PortAudio` library experiences a significant increase in CPU usage in these unusual situations.

When CPU utilization increases, overflow, underflow occurs.

CPU usage was stable in places where there was no problem with frequency meters.

Even other libraries on GitHub have the same problem.

I used libraries of several languages, including C, C++, Go.

Therefore, In my opinion, this problem is not a problem with the `sounddevice` library.

I've been developing software for 8 years, but I've never seen such a case before.

anyway, I decided to move the office, so I think I can close this issue.

Thank you very much for your help.
